feat: Added toggle option for territories on map

derivated opened the pull request:
Added a config setting to choose whether the control key on the main map should act as a hold or a toggle for guild territories, as laid out in issue #1550. If the config setting is set to the default (true), the behavior should be the same as it is currently. If it is set to false, it will act as a toggle. There might be a better way to implement this than what I figured out, but this seems like it will work.

This is my first PR, so I hope I didn't mess something up.

derivated committed:
Added toggle option for territories on map

Added a config setting to choose whether the control key on the main map should act as a hold or a toggle for guild territories, as laid out in issue #1550. If the config setting is set to the default (true), the behavior should be the same as it is currently. If it is set to false, it will act as a toggle. There might be a better way to implement this than what I figured out, but this seems like it will work.

kristofbolyai commented on common/src/main/java/com/wynntils/screens/maps/MainMapScreen.java:
```diff
@@ -262,8 +265,29 @@ private void renderPois(PoseStack poseStack, int mouseX, int mouseY) {
                         .map(PlayerMainMapPoi::new));
 
-        if (KeyboardUtils.isControlDown()) {
+        if (Managers.Feature.getFeatureInstance(MapFeature.class)
```
Let's move this whole logic to key events instead of render:

```
keyDown:
if showTerrs is true and toggleMode is true -> showTerrs is false
otherwise -> showTerrs is true
```

```
keyUp:
if showTerrs is true and toggleMode is false -> showTerrs is false
````

Please override these methods and use this, it is much easier to read.

derivated committed:
Changed implementation of toggling territories on map

Moved logic for the toggle option to the keyPressed and keyReleased methods instead of renderPois

derivated committed:
Merge branch 'main' into main

derivated commented:
How does this look?

kristofbolyai commented on common/src/main/java/com/wynntils/screens/maps/MainMapScreen.java:
```diff
@@ -275,6 +277,36 @@ private void renderPois(PoseStack poseStack, int mouseX, int mouseY) {
 
+    @Override
+    public boolean keyPressed(int keyCode, int scanCode, int modifiers) {
+
```
```suggestion
```

kristofbolyai commented on common/src/main/java/com/wynntils/screens/maps/MainMapScreen.java:
```diff
@@ -275,6 +277,36 @@ private void renderPois(PoseStack poseStack, int mouseX, int mouseY) {
+
+    @Override
+    public boolean keyReleased(int keyCode, int scanCode, int modifiers) {
+
```
```suggestion
```

kristofbolyai committed:
chore: remove empty lines

kristofbolyai commented on common/src/main/resources/assets/wynntils/lang/en_us.json:
```diff
@@ -578,6 +578,8 @@
   "feature.wynntils.map.fastTravelPoiMinZoom.description": "At what zoom level should fast travel pois show? Set this to -1 to always show without fading.",
   "feature.wynntils.map.fastTravelPoiMinZoom.name": "Fast Travel Poi Min Zoom",
+  "feature.wynntils.map.holdGuildMapOpen.description": "Should CTRL hold the guild territories on the map or toggle them?",
+  "feature.wynntils.map.holdGuildMapOpen.name": "Hold Guild Territories Open",
```
```suggestion
  "feature.wynntils.map.holdGuildMapOpen.name": "Hold Control to Show Guild Territories",
```

kristofbolyai commented on common/src/main/resources/assets/wynntils/lang/en_us.json:
```diff
@@ -578,6 +578,8 @@
   "feature.wynntils.map.customPoiMinZoom.name": "Custom Poi Min Zoom",
   "feature.wynntils.map.fastTravelPoiMinZoom.description": "At what zoom level should fast travel pois show? Set this to -1 to always show without fading.",
   "feature.wynntils.map.fastTravelPoiMinZoom.name": "Fast Travel Poi Min Zoom",
+  "feature.wynntils.map.holdGuildMapOpen.description": "Should CTRL hold the guild territories on the map or toggle them?",
```
```suggestion
  "feature.wynntils.map.holdGuildMapOpen.description": "Should Control need to be held to open guild territories on the map. If this is disabled, control will act as a toggle.",
```

kristofbolyai committed:
fix: fix i18n
